### PR TITLE
Polish authentication and documentation

### LINKED ISSUE #35: Make authentication optional?
Hi, would it be possible to make authentication optional? We run cromwell in a walled environment and would prefer not to deal with authentication. Currently if no form of authentication is passed to `start_workflow` this exception is raised from `harmonize_credentials:

```
 File "envs/pipeflow/src/cromwell-tools/cromwell_tools/cromwell_tools.py", line 184, in start_workflow
    auth, headers = _get_auth_credentials(cromwell_user=user, cromwell_password=password, caas_key=caas_key)
  File "envs/pipeflow/src/cromwell-tools/cromwell_tools/cromwell_tools.py", line 59, in _get_auth_credentials
    cromwell_user, cromwell_password = harmonize_credentials(secrets_file, cromwell_user, cromwell_password)
  File "envs/pipeflow/src/cromwell-tools/cromwell_tools/cromwell_tools.py", line 43, in harmonize_credentials
    raise ValueError('One form of cromwell authentication must be provided, please pass '
ValueError: One form of cromwell authentication must be provided, please pass either cromwell_user and cromwell_password or a secrets_file.
```

### LINKED ISSUE #53: oauth2client is deprecated
According to https://google-auth.readthedocs.io/en/latest/oauth2client-deprecation.html, the `oauth2client` lib that we are using in cromwell-tools has been deprecated in favor of [`google-oauth`](https://github.com/googleapis/google-auth-library-python), for both security and usability concerns, we need to update cromwell-tools with the new dependency.

### PR DESCRIPTION
**Pull request by rexwangcc**
This PR focused on:
- [x] Updating cromwell-tools to address the deprecation of `oauth2client`, fixes #53
- [x] Making authentication optional, fixes #35 
- [x] Cleaning up the readme and documentation generated by Sphinx.

---
Please ensure the following when opening a PR:
- [x] Added or updated tests
- [x] Updated documentation
- [x] Applied style guidelines
- [x] Considered generalizability beyond our own use case

### CODE REVIEW

File: cromwell_tools/cromwell_auth.py [hunk 1]
```diff
@@ -136,32 +145,47 @@ def from_user_password(cls, username, password, url):
...
+                with CaaS or any Cromwell instances that using OAuth.
 
         Returns:
             CromwellAuth: An instance of this auth helper class with proper credentials for authenticating with
                 Cromwell or CaaS.
         """
         # verify only one credential provided
         credentials = {
-            "service_account_key": True if service_account_key else False,
+            "service_account_key": True if all((service_account_key, url)) else False,
```
Comment (danxmoran) [inline cromwell_tools/cromwell_auth.py]: Could this be simplified to just `all((service_account_key, url))`?

File: cromwell_tools/cromwell_auth.py [hunk 2]
```diff
@@ -136,32 +145,47 @@ def from_user_password(cls, username, password, url):
...
             "secrets_file": True if secrets_file else False,
-            "user_password": True if all((username, password, url)) else False
+            "user_password": True if all((username, password, url)) else False,
```
Comment (danxmoran) [inline cromwell_tools/cromwell_auth.py]: Same question

File: cromwell_tools/cromwell_auth.py [hunk 3]
```diff
@@ -136,32 +145,47 @@ def from_user_password(cls, username, password, url):
...
+            "no_auth": True if (service_account_key, secrets_file, username, password).count(None) == 4 and
```
Comment (danxmoran) [inline cromwell_tools/cromwell_auth.py]: Is there an `any` to match `all`? If so, could this be `!any((service_account_key, secrets_file, username, password))`?

File: cromwell_tools/cromwell_auth.py (hunk 1 above)
Comment (rexwangcc) [reply cromwell_tools/cromwell_auth.py]: It could, and in fact, it's better! Will do

File: cromwell_tools/cromwell_auth.py (hunk 3 above)
Comment (rexwangcc) [reply cromwell_tools/cromwell_auth.py]: Good point, `not any((service_account_key, secrets_file, username, password))` should work!

**Pull request merged**